### ISSUE TITLE
Show OSMF membership status

### ISSUE THREAD

**Issue by pnorman** [openstreetmap/openstreetmap-website#1776]
The Membership Working Group is working on showing that someone is an OSMF member on their osm.org profile

We would like a checkbox that shows in a user's settings, giving them the option to display that they are an OSMF member.
If the checkbox is checked and they are still a member, it displays on their profile page.

@datendelphin, you've identified that we can export from CiviCRM. What format does it generate?

There is also an operations question about regularly running the CiviCRM export and getting it to the website, but I figure that's best addressed after the facility to use that data is added.

**Comment by tomhughes** [comment]
Just to repeat what I said when we discussed this on IRC my main concern here is that I don't think we have any way to accurately match OSMF members to OSM accounts.

As I understand it we ask people for their account name at signup but then never make any attempt to get updates so we may well wind up attaching these flags to the wrong accounts.

**Comment by datendelphin** [comment]
It generates json.
I hope that will motivate people to update it. But hopefully most will not change their user name anyway. It is planned to be opt-in, so in the worst case a wrong user has the option to show himself as being an OSMF member even if he/she is not. Which I'm not too much concerned about.

**Comment by tomhughes** [comment]
Where is the opt in going to be exactly/? in CiviCRM?

**Comment by pnorman** [comment]
> Where is the opt in going to be exactly? in CiviCRM?

It'll show as a checkbox in the user profile, giving them the option of displaying they're an OSMF member. The worst-case scenario where an OSMF member changes OSM username and someone else takes their original username results in the wrong person seeing the option.

**Comment by tomhughes** [comment]
You mean the user profile on the OSM web site? This may sound odd, but that's very OSM specific which creates a problem for all the other people that use our code for other things.

**Comment by dieterdreist** [comment]
sent from a phone

> On 3. Mar 2018, at 01:51, Paul Norman <notifications@github.com> wrote:
> 
> It'll show as a checkbox in the user profile, giving them the option of displaying they're an OSMF member.


can multiple accounts be linked to the same OSMF member?

**Comment by mikelmaron** [comment]
> This may sound odd, but that's very OSM specific which creates a problem for all the other people that use our code for other things.

There may be a more generic way to approach this, which would allow OSM users to add information from multiple external services to their user profile. OSMF membership, HDYC details, etc.

* Services provide an endpoint that takes an OSM userid, return javascript which writes HTML
* OSM instance is configured to make external services available to users
* OSM user can choose options of which external services to include in their user profile
* User profile page makes request to services endpoints from the browser, executes to display result.

**Comment by pnorman** [comment]
> You mean the user profile on the OSM web site? This may sound odd, but that's very OSM specific which creates a problem for all the other people that use our code for other things.

Yes, this feature would be useless to other people running the rails port. Aside from text and images that reference OSM by name, this might be the first feature which is of **no** use to other deployments. Everything else I can think of has a marginal value to another deployment if they're willing to go to the effort of setting it up. Frequently they don't, e.g. OpenHistoricalMap is unlikely to set up routing soon.

There's always been a tension between this being the software that powers part of osm.org and it being general-purpose software that can be used for crowd-sourced geodata projects by others (e.g. OpenHistoricalMap, OpenGeoFiction, government agencies, etc). Because the two are incompatible in this case, I guess it needs a maintainer call on which has to give.

-----

> can multiple accounts be linked to the same OSMF member?

No, this would require changes on the CiviCRM side which are out of scope for this tracker. Multiple OSMF members inputting the same account isn't an issue because the result of each member is the same - display the option.

-----

> > This may sound odd, but that's very OSM specific which creates a problem for all the other people that use our code for other things.

> There may be a more generic way to approach this, which would allow OSM users to add information from multiple external services to their user profile. OSMF membership, HDYC details, etc.

Because HDYC, etc are all also specific to this instance of the rails port and the data it creates this wouldn't help the problem.

> Services provide an endpoint that takes an OSM userid, return javascript which writes HTML

This wouldn't work here, we can't make the membership lists publicly available, which an API would do. This issue is about a route where the only personal information disclosed would be to the OSM account holder, who the OSMF member has said themselves.

**Comment by datendelphin** [comment]
Maybe we could make this feature optional with some configuration, to ease the usage by other parties.

I understand why this issue produced the proposal by mikelmaron, but I will not take that on. That is so much more complicated and definitely not feasible for me.

**Comment by woodpeck** [comment]
Frankly, I think that we have enough on our plates at OSM already and while it's ok to make the openstreetmap-website software reusable for other purposes, it's also ok if whoever wants to reuse it has to make *some* changes or adapt it to their needs. I don't view openstreetmap-website as "the project that develops software for various open mapping communities around the globe" and when faced with the decision of adding a simple feature that would be unusable to others who install this software, or not adding the feature because doing it in a generic way would over-tax our limited resources, I'd clearly favour the "add an OSM-only feature" approach.

**Comment by pnorman** [comment]
> it's also ok if whoever wants to reuse it has to make some changes or adapt it to their needs

Thinking about it some more, if someone reusing the software left code for this feature alone it would probably work fine. With no list of OSMF members in their instance their users would just never see the checkbox.
